SourceKitStressTester/Sources/StressTester/SourceKitDocument.swift

```diff
@@ -227,15 +233,46 @@ class SourceKitDocument {
   }
 
   func codeComplete(offset: Int, expectedResult: ExpectedResult?) throws -> (request: RequestInfo, response: SourceKitdResponse, instructions: Int) {
-    let request = SourceKitdRequest(uid: .request_CodeComplete)
-
-    request.addParameter(.key_SourceFile, value: args.forFile.path)
-    if let sourceState = sourceState {
-      request.addParameter(.key_SourceText, value: sourceState.source)
+    let location = converter!.location(for: AbsolutePosition(utf8Offset: offset))
+
+    /// Build the codecomplete.open or codecomplete.close request.
+    /// The two requests share the exact same options except for the request name.
+    func buildRequest(request: SourceKitdUID) -> SourceKitdRequest {
+      let request = SourceKitdRequest(uid: request)
+      request.addParameter(.key_SourceFile, value: args.forFile.path)
+      request.addParameter(.key_Name, value: args.forFile.path)
+      if let sourceState = sourceState {
+        request.addParameter(.key_SourceText, value: sourceState.source)
+      }
+      request.addParameter(.key_Offset, value: offset)
+      request.addParameter(.key_Line, value: location.line!)
+      request.addParameter(.key_Column, value: location.column!)
+      request.addCompilerArgs(args.sourcekitdArgs)
+
+      let completionOptions = request.addDictionaryParameter(.key_CodeCompleteOptions)
+      // Don't hide any results because otherwise checking if the expected result is contained will fail
+      completionOptions.add(.key_HideLowPriority, value: 0)
+      completionOptions.add(.key_HideByName, value: 0)
+      completionOptions.add(.key_HideUnderscores, value: 0)
+
+      // Set expected results to 200 to avoid inflating the time measurements by serialization time.
+      // To make sure that the expected result is in the results, filter by its base name.
+      completionOptions.add(.key_RequestLimit, value: 200)
+      if let expectedResult = expectedResult {
+        completionOptions.add(.key_FilterText, value: expectedResult.name.base)
```

Comment:
This would give fewer completions than usual, but I guess the assumption is that the serialization time of 0-200 results is negligible anyway?

Reply:
Yes, exactly.